File: 402/app/src/main/java/com/example/a402/MainActivity.java
package com.example.a402;

import android.graphics.Color;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private View rootView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        rootView = findViewById(android.R.id.content);

        Button buttonYellow = findViewById(R.id.buttonYellow);
        Button buttonBlue = findViewById(R.id.buttonBlue);
        Button buttonRed = findViewById(R.id.buttonRed);
        Button buttonGreen = findViewById(R.id.buttonGreen);

        buttonYellow.setOnClickListener(v -> changeBackgroundColor(Color.YELLOW));
        buttonBlue.setOnClickListener(v -> changeBackgroundColor(Color.BLUE));
        buttonRed.setOnClickListener(v -> changeBackgroundColor(Color.RED));
        buttonGreen.setOnClickListener(v -> changeBackgroundColor(Color.GREEN));
    }

    private void changeBackgroundColor(int color) {
        rootView.setBackgroundColor(color);
    }
}

File: 21_05/app/src/main/java/com/example/a21_05/MainActivity.kt
package com.example.a21_05

import android.annotation.SuppressLint
import android.graphics.Color
import android.os.Bundle
import android.view.MotionEvent
import android.widget.Button
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    @SuppressLint("ClickableViewAccessibility")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)

        val button = findViewById<Button>(R.id.button)

        button.setOnTouchListener { view, event ->
            when (event.action) {
                MotionEvent.ACTION_DOWN -> {
                    view.setBackgroundColor(Color.GREEN)
                    true
                }
                MotionEvent.ACTION_UP -> {
                    view.setBackgroundColor(Color.BLUE)
                    true
                }
                MotionEvent.ACTION_MOVE -> {
                    view.x = event.rawX - view.width / 2
                    view.y = event.rawY - view.height / 2
                    view.setBackgroundColor(Color.YELLOW)
                    true
                }
                else -> {false}
            }
        }
    }
}